services/checks/checks.go

```diff
@@ -1116,13 +1020,14 @@ func (s *Service) filterSupportedChecks(checks []check.Check) []check.Check {
 }
 
 // updateChecks update service checks filed value under mutex.
-func (s *Service) updateChecks(mySQLChecks, postgreSQLChecks, mongoDBChecks map[string]check.Check) {
+func (s *Service) updateChecks(checks []check.Check) {
 	s.cm.Lock()
 	defer s.cm.Unlock()
 
-	s.mySQLChecks = mySQLChecks
-	s.postgreSQLChecks = postgreSQLChecks
-	s.mongoDBChecks = mongoDBChecks
+	s.checks = make(map[string]check.Check)
```

Comment:
Here we `replace` previous checks with the new, so we need to throw away everything that we have. The best way to do that is just to create a new map. Otherwise, you should iterate over an existing map and remove elements one by one. I'm pretty sure that it will work slower than just new map allocation.